`theme` crate spring cleaning

iamnbutler opened the pull request:
This PR does some spring cleaning on the `theme` crate:

- Removed two unused stories and the story dep
- Removed the `one` theme family (from the `theme` crate, not the app), this is now `zed_default_themes`.
  - This will hopefully remove some confusion caused by this theme we started in rust but didn't end up using
- Removed `theme::prelude` (it just re-exported scale colors, which we don't use outside `theme`)
- Removed completely unused `zed_pro` themes (we started on these during the gpui2 port and didn't finish them.)

Release Notes:

- N/A

iamnbutler committed:
Update theme.rs

iamnbutler committed:
Update default_colors.rs

iamnbutler committed:
Update settings.rs

iamnbutler committed:
Update font_family_cache.rs

iamnbutler committed:
Update default_theme.rs

iamnbutler committed:
Update accents.rs

iamnbutler committed:
Allow missing docs on styles for now

iamnbutler committed:
Update theme.rs

iamnbutler committed:
Allow missing schema docs for now

iamnbutler committed:
Update registry.rs

iamnbutler committed:
Update scale.rs

iamnbutler committed:
Update settings.rs

iamnbutler committed:
Add missing story docs

iamnbutler committed:
Fix typos

iamnbutler committed:
Remove unused utilities

iamnbutler committed:
Remove unused & outdated doc

iamnbutler committed:
Merge branch 'main' into tidy-theme-crate

iamnbutler committed:
Remove unused stories

These weren't in the story selector

iamnbutler committed:
Remove unused zed default themes

iamnbutler committed:
Move default themes to crate root

iamnbutler committed:
Remove unused theme prelude

iamnbutler committed:
Remove confusing "one" theme family

iamnbutler committed:
Remove unused story dep

maxdeviant commented on crates/theme/src/fallback_themes.rs:
```diff
@@ -7,21 +7,21 @@ use crate::{
     ThemeFamily {
-        id: "one".to_string(),
-        name: "One".into(),
+        id: "zed default".to_string(),
```
```suggestion
        id: "zed-default".to_string(),
```

maxdeviant committed:
Update crates/theme/src/fallback_themes.rs

maxdeviant committed:
Merge branch 'main' into tidy-theme-crate

maxdeviant committed:
Amend commit

maxdeviant committed:
Remove unneeded `Default` impls

We aren't using these anywhere, and I don't think the `Default` semantics map well to themes,
where the user's choice needs to be considered

maxdeviant reviewed (approved):
Looks good 👍🏻
